message_sender/tests.py

```diff
@@ -3175,6 +3175,60 @@ def test_requeue(self):
         self.assertEqual(OutboundSendFailure.objects.all().count(), 0)
 
 
+class TestFailedMsisdnLookUp(TestCase):
+    def add_identity_search_response(self, msisdn, identity, count=1):
+        msisdn = msisdn.replace("+", "%2B")
+        results = [
+            {
+                "id": identity,
+                "version": 1,
+                "details": {
+                    "default_addr_type": "msisdn",
+                    "addresses": {"msisdn": {msisdn: {}}},
+                },
+            }
+        ] * count
+        response = {"next": None, "previous": None, "results": results}
+        qs = "?details__addresses__msisdn=%s" % msisdn
+        responses.add(
+            responses.GET,
+            "%s/identities/search/%s" % (settings.IDENTITY_STORE_URL, qs),  # noqa
+            json=response,
+            status=200,
+            match_querystring=True,
+        )
+
+    @responses.activate
+    def test_fire_failed_msisdn_lookup(self):
+        """
+        trigger a webhook if there is no to_addr in the identity
+        """
+        send_message = SendMessage()
+        self.add_identity_search_response("", str(uuid.uuid4()))
+
+        responses.add(
+            method=responses.POST,
+            url="http://example.com",
+            json={"identity": [{"to_addr": None}]},
+            status=200,
+        )
+
+        user = User.objects.create_user("test")
+        hook = Hook.objects.create(
+            event="identity.no_address", target="http://webhook", user=user
+        )
+
+        responses.add(method=responses.POST, url="http://webhook", json={}, status=200)
+
+        self.assertEqual(send_message.run(str(uuid.uuid4())), None)
+        self.assertEqual(send_message.fire_failed_msisdn_lookup(None), None)
```

Comment:
I don't think that we want to still be calling this directly. The `send_message.run` call above should be calling this function.

Reply:
so I just call the 1st assert and not the 2nd?

Reply:
Yup, that would be correct.

The call to `run` should do a call to `fire_failed_msisdn_lookup`, so by calling it manually, we'll be calling it twice.